### PR TITLE
526 update config schema mg

### PR DESCRIPTION
**Pull request by markgreenburg**
Updates config to make use of the new schema.

### CODE REVIEW

**Comment by markgreenburg** [comment]
NOTE: Needs https://github.com/department-of-veterans-affairs/vets-json-schema/pull/208 to be merged first, then the schemaform dependency needs to be updated before merging.

File: src/applications/disability-benefits/526EZ/containers/ConfirmationPage.jsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ class ConfirmationPage extends React.Component {
             <strong>Conditions claimed</strong>
             <br/>
             <ul className="disability-list">
-              {selectedDisabilities.map((disability, i) => {
+              {disabilities.map((disability, i) => {
```
Comment (cehsu) [inline src/applications/disability-benefits/526EZ/containers/ConfirmationPage.jsx]: Will these be filtered to only display the selected disabilities?

**Review by cehsu — APPROVED** [review]
LGTM. I had one question about showing the selected disabilities on the confirmation page.

File: src/applications/disability-benefits/526EZ/config/form.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 import _ from '../../../../platform/utilities/data';
 
-import { omitRequired } from '../../../common/schemaform/helpers';
+// import { omitRequired } from '../../../common/schemaform/helpers';
 
 import fullSchema526EZ from 'vets-json-schema/dist/21-526EZ-schema.json';
 // NOTE: Easier to run schema locally with hot reload for dev
-// import fullSchema526EZ from '/local/path/vets-json-schema/dist/21-526EZ-schema.json';
+// import fullSchema526EZ from '/Users/adhocteam/Sites/vets-json-schema/dist/21-526EZ-schema.json';
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/config/form.js]: Maybe this change doesn't make sense? :man_shrugging:

File: src/applications/disability-benefits/526EZ/config/form.js [hunk 3]
```diff
@@ -103,6 +104,16 @@ const treatments = ((treatmentsCommonDef) => {
 
 })(treatmentsSchema);
 
+const initialData = ((prefill) => {
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/config/form.js]: This seems more complicated than it needs to be, no? Why the immediately invoked function?

File: src/applications/disability-benefits/526EZ/containers/ConfirmationPage.jsx (hunk 1 above)
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/containers/ConfirmationPage.jsx]: Yeah, that's a good idea. It's not really part of this PR, but it's an easy enough change.

```js
disabilities.filter(d => d['view:selected']).map(...)
```

File: src/applications/disability-benefits/526EZ/helpers.jsx [hunk 4]
```diff
@@ -347,12 +348,14 @@ export const veteranInformationViewField = ({ formData }) => {
  */
 export const disabilityOption = ({ diagnosticCode, name, ratingPercentage }) => {
   // May need to throw an error to Sentry if any of these doesn't exist
+  // A valid rated disability *can* have a rating percentage of 0%
+  const showRatingPercentage = !!(ratingPercentage || ratingPercentage === 0);
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/helpers.jsx]: This works, but can be simplified with `Number.isInteger(ratingPercentage)`.

File: src/applications/disability-benefits/526EZ/tests/containers/ConfirmationPage.unit.spec.jsx [hunk 5]
```diff
@@ -21,10 +21,14 @@ describe('Disability Benefits 526EZ <ConfirmationPage>', () => {
...
     );
 
-    expect(tree.find('.confirmation-page-title').text()).to.equal('Claim submitted');
-    expect(tree.find('span').at(1).text().trim()).to.equal('For Sally  Alphonse');
-    expect(tree.find('p').at(0).text()).to.contain('We usually process claims within 103 days.');
-    expect(tree.find('p').at(1).text()).to.contain('We may contact you if we have questions or need more information. You can print this page for your records.');
-    expect(tree.find('.disability-list').text()).to.contain('PTSD');
+    expect(tree.find('.confirmation-page-title').render().text()).to.equal('Claim submitted');
+    expect(tree.find('span').at(1)
+      .render()
```
Comment (cvalarida) [inline src/applications/disability-benefits/526EZ/tests/containers/ConfirmationPage.unit.spec.jsx]: Idle curiosity: Why break this one out onto separate lines?

**Review by cvalarida — APPROVED** [review]
Looks good. There are a few opportunities to reduce complexity, but otherwise, it looks good.

File: src/applications/disability-benefits/526EZ/config/form.js (hunk 2 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/config/form.js]: That's not what was supposed to happen...

File: src/applications/disability-benefits/526EZ/helpers.jsx (hunk 4 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: Not supported in IE, though `babel-polyfill` probably takes care of this.

File: src/applications/disability-benefits/526EZ/tests/containers/ConfirmationPage.unit.spec.jsx (hunk 5 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/tests/containers/ConfirmationPage.unit.spec.jsx]: Hah, ask the linter 😛

File: src/applications/disability-benefits/526EZ/containers/ConfirmationPage.jsx (hunk 1 above)
Comment (markgreenburg) [reply src/applications/disability-benefits/526EZ/containers/ConfirmationPage.jsx]: Good thinking; probably needs its own issue.

File: src/applications/disability-benefits/526EZ/config/form.js (hunk 3 above)
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/config/form.js]: :man_shrugging:

File: src/applications/disability-benefits/526EZ/helpers.jsx (hunk 4 above)
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/helpers.jsx]: Yeah, I think we can trust `babel-polyfill` to do that. The whole babel multiverse confuses me sometimes, but I did see a closed issue regarding this.

File: src/applications/disability-benefits/526EZ/tests/containers/ConfirmationPage.unit.spec.jsx (hunk 5 above)
Comment (cvalarida) [reply src/applications/disability-benefits/526EZ/tests/containers/ConfirmationPage.unit.spec.jsx]: :roll_eyes:

**Pull request merged**